src/main/java/net/dries007/tfc/common/blocks/rock/RockSpikeBlock.java

```diff
@@ -103,52 +130,40 @@ else if (belowState.isFaceSturdy(level, belowPos, Direction.UP))
             return;
         }
 
-        // No support below, try above
-        BlockPos abovePos = pos.above();
-        BlockState aboveState = level.getBlockState(abovePos);
-        if (aboveState.getBlock() == this && aboveState.getValue(PART).isLargerThan(state.getValue(PART)))
+        if (checkAbove)
         {
-            // Larger spike above. Tick to ensure that it is supported
-            level.scheduleTick(abovePos, this, 1);
-            return;
-        }
-        else if (aboveState.isFaceSturdy(level, abovePos, Direction.DOWN))
-        {
-            // Full block above, this is supported
-            return;
+            // No support below, try above
+            final BlockPos abovePos = pos.above();
+            final BlockState aboveState = level.getBlockState(abovePos);
+            if (aboveState.getBlock() == this && aboveState.getValue(PART).isLargerThan(state.getValue(PART)))
+            {
+                // Larger spike above. Tick to ensure that it is supported
+                level.scheduleTick(abovePos, this, 1);
+                return;
+            }
+            else if (aboveState.isFaceSturdy(level, abovePos, Direction.DOWN))
```

Comment:
this seems weird because of how baby spikes don't need a full face.

Reply:
They do though? I'm not sure what you're talking about (this logic is also unchanged - just moved). There's not any functionality bug that I'm aware of, nor that the tests have caught.